exporter/otlp-common/lib/opentelemetry/exporter/otlp/common.rb

```diff
@@ -68,8 +71,67 @@ def as_etsr(span_data) # rubocop:disable Metrics/MethodLength
           )
         end
 
+        def as_encoded_emsr(metric_data)
+          Opentelemetry::Proto::Collector::Metrics::V1::ExportMetricsServiceRequest.encode(as_emsr(metric_data))
+        end
+
+        def as_emsr(metric_data)
+          Opentelemetry::Proto::Collector::Metrics::V1::ExportMetricsServiceRequest.new(
+            resource_metrics: metric_data
+              .group_by(&:resource)
+              .map do |resource, metric_datas|
+                Opentelemetry::Proto::Metrics::V1::ResourceMetrics.new(
+                  resource: Opentelemetry::Proto::Resource::V1::Resource.new(
+                    attributes: resource.attribute_enumerator.map { |key, value| as_otlp_key_value(key, value) }
+                  ),
+                  instrumentation_library_metrics: metric_datas
+                  .group_by(&:instrumentation_library)
+                  .map do |il, mds|
+                    Opentelemetry::Proto::Metrics::V1::InstrumentationLibraryMetrics.new(
+                      instrumentation_library: Opentelemetry::Proto::Common::V1::InstrumentationLibrary.new(
+                        name: il.name,
+                        version: il.version
+                      ),
+                      metrics: mds.group_by.map do |metric_stream|
```

Comment:
what does the `group_by` do here?

Reply:
Some left over from previous iterations, I'll clean this up.

I'll be removing all the export work from this PR, and push it up as an entirely separate PR for review.